api/client/commands.go

```diff
@@ -2183,7 +2183,7 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 	//create the container
 	stream, statusCode, err := cli.call("POST", "/containers/create?"+containerValues.Encode(), mergedConfig, false)
 	//if image not found try to pull it
-	if statusCode == 404 {
+	if statusCode == 404 && strings.Contains(err.Error(), config.Image) {
```

Comment:
is there any chance the error has `config.Image` in the string, but isn't a "not found" scenario?  I assume it shouldn't be as 404 should be reserved for that, but it is clear that we are getting 404's for errors not specifically related to "image not found" (of the image used in create, not other "image names" that may appear on the command line).